Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,229 @@
+            {
+                product.SeoSlug = patchDoc.SeoSlug;
+            }
+
+            if (patchDoc.SeoDescription != null)
+            {
+                product.SeoDescription = patchDoc.SeoDescription;
+            }
+
+            var saved = await db.Save();
+
+            return saved ? Ok(product) : Problem();
+        }
+
+
+        [HttpPut("{id}/publish")]
+        public async Task<IActionResult> PublishProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            product.IsDraft = false;
+
+            var saved = await db.Save();
+            return saved ? Ok(product) : Problem();
+        }
+
+
+        [HttpPost("{id}/images")]
+        public async Task<IActionResult> UploadProductImages(string id, [FromForm] ICollection<IFormFile> imageFiles)
+        {
+            if (imageFiles != null)
+            {
+                var product = await db.Products.QueryOne(p => p.Id == id);
+
+                if (product == null)
+                {
+                    return NotFound();
+                }
+
+                var productImagesToDelete = await db.ProductImages.QueryMany(pi => pi.ProductId == id);
+                db.ProductImages.RemoveRange(productImagesToDelete);
+                await imageService.SafeDelete(productImagesToDelete);
+
+
+                foreach (var imageFile in imageFiles)
+                {
+                    if (IsImageFile(imageFile))
+                    {
+                        var fileId = Guid.NewGuid().ToString() + Path.GetExtension(imageFile.FileName);
+
+                        using (var stream = imageFile.OpenReadStream())
+                        {
+                            var uploadedFile = await storage.Upload(fileId, stream);
+
+                            if (uploadedFile != null)
+                            {
+                                var productImage = new ProductImage(uploadedFile.Provider, uploadedFile.Bucket, uploadedFile.Key, id);
+                                db.ProductImages.Add(productImage);
+                            }
+                        }
+                    }
+                }
+
+                var saved = await db.Save();
+                return saved ? Ok() : Problem();
+            }
+
+            return Ok();
+        }
+
+
+        public static bool IsImageFile(IFormFile file)
+        {
+            if (file == null || string.IsNullOrEmpty(file.FileName) || file.Length == 0)
+            {
+                return false;
+            }
+
+            var fileExtension = Path.GetExtension(file.FileName).ToLower();
+            string[] photoExtensions = { ".jpg", ".jpeg", ".png", ".gif", ".bmp" };
+
+            return photoExtensions.Contains(fileExtension);
+        }
+
+
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteProduct(string id)
```

Comment:
You should check for user. Add [Authorize]. Get uid by User.FindFirst(Jwt.Uid). Check that shop is owned by user.